packages/mgt-components/src/components/mgt-todo/mgt-todo.ts

```diff
@@ -152,242 +126,273 @@ export class MgtTodo extends MgtTasksBase {
+            <div dir=${this.direction} class="Task NewTask Incomplete">
+              ${newTaskDetails}
+            </div>
+        `
+          : html``
+      }  
+     `;
+  };
 
   /**
-   * Render the header part of the component.
+   * Handle a change in taskList.
    *
    * @protected
-   * @returns
+   * @param {CustomEvent} e
+   * @returns {TemplateResult}
    * @memberof MgtTodo
    */
-  protected renderHeaderContent(): TemplateResult {
-    if (this.isLoadingState) {
-      return html`
-        <div class="header__loading"></div>
-      `;
+
+  protected handleSelectionChanged = (e: CustomEvent<TodoTaskList>) => {
+    const list: TodoTaskList = e.detail;
+    this.currentList = list;
+    void this.loadTasks(list);
+  };
+
+  /**
+   * Render a task in the list.
+   *
+   * @protected
+   * @param {TodoTask} task
+   * @returns {TemplateResult}
+   * @memberof MgtTodo
+   */
+  protected renderTask = (task: TodoTask) => {
+    const context = { task, list: this.currentList };
+
+    if (this.hasTemplate('task')) {
+      return this.renderTemplate('task', context, task.id);
     }
 
-    const lists = this._lists || [];
-    const currentList = this._currentList;
-    const targetId = this.targetId;
-    let listSelect: TemplateResult;
-
-    if (targetId && lists.length) {
-      const list = lists.find(l => l.id === targetId);
-      if (list) {
-        listSelect = html`
-          <span class="PlanTitle">
-            ${list.displayName}
-          </span>
-        `;
-      }
-    } else if (currentList) {
-      const listOptions = {};
-      for (const l of lists) {
-        listOptions[l.displayName] = () => this.loadTaskList(l);
-      }
+    let taskDetailsTemplate = null;
 
-      listSelect = mgtHtml`
-        <mgt-arrow-options .value="${currentList.displayName}" .options="${listOptions}"></mgt-arrow-options>
+    const taskDueTemplate = task.dueDateTime
+      ? html`
+        <span class="TaskCalendar">${getSvg(SvgIcon.Calendar)}</span>
+        <span class="TaskDueDate">${getDateString(new Date(task.dueDateTime.dateTime))}</span>
+      `
+      : html``;
+
+    if (this.hasTemplate('task-details')) {
+      taskDetailsTemplate = this.renderTemplate('task-details', context, `task-details-${task.id}`);
+    } else {
+      taskDetailsTemplate = html`
+      <div class="TaskDetails">
+        <div class="Title">${task.title}</div>
+        <div class="TaskDue">${taskDueTemplate}</div>
+        <fluent-button class="TaskDelete"
+          @click="${() => this.removeTask(task.id)}"
+          aria-label="${this.strings.deleteTaskLabel}">
+          ${getSvg(SvgIcon.Delete)}
+        </fluent-button>
+      </div>
       `;
     }
 
+    const taskClasses = classMap({
+      ReadOnly: this.readOnly,
+      Task: true
+    });
+
     return html`
-      <span class="TitleCont">
-        ${listSelect}
-      </span>
+      <fluent-checkbox id=${task.id} class=${taskClasses} ?checked=${false} @click="${() =>
```

Comment:
```suggestion
      <fluent-checkbox id=${task.id} class=${taskClasses} @click="${() =>
```

This is always not checked.